src/Umbraco.Web/Strategies/RelateOnTrashHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Auditing;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Web.Strategies
+{
+    public sealed class RelateOnTrashHandler : ApplicationEventHandler
```

Comment:
Thanks @Shazwazza - as suggested, I only put this in the `Umbraco.Web.Strategies` namespace because `RelateOnCopyHandler` was already there.

I agree about having relations created outside a web-context.